lib/SILGen/SILGenProlog.cpp

```diff
@@ -97,12 +97,17 @@ class EmitBBArguments : public CanTypeVisitor<EmitBBArguments,
 
   ManagedValue visitType(CanType t, bool isInOut) {
     auto argType = SGF.getLoweredType(t);
-    if (isInOut)
-      argType = SILType::getPrimitiveAddressType(argType.getASTType());
 
     // Pop the next parameter info.
     auto parameterInfo = parameters.front();
     parameters = parameters.slice(1);
+
+    if (parameterInfo.isFormalIndirect()) {
```

Comment:
Mark this as SWIFT_ENABLE_TENSORFLOW?

Can you also add a comment explaining why you need to do this?